### PR TITLE
[dice] Refactor the `dice` library to support multiple implementations

### PR DESCRIPTION
**Pull request by tommychiu-github**
To support mutliple DICE implementation (X.509 with ASN.1, and CWT with CBOR), it'll require to refactor current dice lib implementation.
1. Add a `dice_cwt` lib (implementation details to be added later)
2. Levaraging the third_party `open_dice` library to generate UDS_Pub structure

### CODE REVIEW

File: sw/device/silicon_creator/lib/cert/cbor_helper.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#include "include/dice/cbor_writer.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CBOR_CHECK_OVERFLOWED_AND_RETURN(p) \
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/cbor_helper.h:16]: undef this at the end of this file? https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros

File: sw/device/silicon_creator/lib/cert/BUILD [hunk 2]
```diff
@@ -112,6 +112,42 @@ cc_library(
...
+        "//sw/device/silicon_creator/lib/base:util",
+        "//sw/device/silicon_creator/lib/cert:cdi_0_template_library",
+        "//sw/device/silicon_creator/lib/cert:cdi_1_template_library",
+        "//sw/device/silicon_creator/lib/cert:uds_template_library",
+        "//sw/device/silicon_creator/lib/drivers:hmac",
+        "//sw/device/silicon_creator/lib/drivers:keymgr",
+        "//sw/device/silicon_creator/lib/drivers:lifecycle",
+        "//sw/device/silicon_creator/lib/drivers:otp",
+        "//sw/device/silicon_creator/lib/sigverify:ecdsa_p256_key",
+        "//sw/device/silicon_creator/manuf/lib:flash_info_fields",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/BUILD:147]: can you double check the deps you have here, I don't think you need a lot of these

File: sw/device/silicon_creator/lib/cert/cbor_helper.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CERT_CBOR_HELPER_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CERT_CBOR_HELPER_H_
+
+#include "include/dice/cbor_writer.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/cbor_helper.h:14]: This only makes sense if we plan to unittest this with a C++ test code. I think you can remove for now.

File: sw/device/silicon_creator/lib/cert/dice_cwt.c [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  *tbs_cert_size = CborOutSize(pCborOut);
+
+  return kErrorOk;
+}
+
+rom_error_t dice_cdi_0_cert_build(hmac_digest_t *rom_ext_measurement,
+                                  uint32_t rom_ext_security_version,
+                                  cert_key_id_pair_t *key_ids,
+                                  ecdsa_p256_public_key_t *cdi_0_pubkey,
+                                  uint8_t *cert, size_t *cert_size) {
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/cert/dice_cwt.c:154]: Can you add `// TODO(lowRISC/opentitan:#24281): implement body` comments in these functions that are not yet implemented? (https://opentitan.org/book/doc/contributing/detailed_contribution_guide/index.html?highlight=todo#what-about-todo-comments)

**Comment by timothytrippel** [comment]
@tommychiu-github for future reference, please update the title and descriptions of your PRs to be more descriptive and formatted like commit message. Thanks!

File: sw/device/silicon_creator/lib/cert/dice_cwt.c [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+// Values
+const int64_t kCoseKeyAlgEcdsa256 = -7;
+const int64_t kCoseEc2CrvP256 = 1;
+const int64_t kCoseKeyKtyEc2 = 2;
+
+// UDS (Creator) attestation key diverisfier constants.
+// Note: versions are always set to 0 so these keys are always valid from the
+// perspective of the keymgr hardware.
+const sc_keymgr_diversification_t kUdsKeymgrDiversifier = {
```
[the remark this answers is not in the document]
Comment (tommychiu-github) [reply sw/device/silicon_creator/lib/cert/dice_cwt.c:52]: Good idea. Will push a commit for that later.

**Comment by tommychiu-github** [comment]
> @tommychiu-github for future reference, please update the title and descriptions of your PRs to be more descriptive and formatted like commit message. Thanks!

Updated.

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**